File: API/lb_clientAPI_sources/lb/api/beans/LetterDeliveryType.java
package lb.api.beans;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import lb.api.connectors.SessionClient;
import lb.api.exceptions.RemoteException;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class LetterDeliveryType {
	
	protected String id;
	protected String code;
	protected String name;

	protected static List<LetterDeliveryType> letterDeliveryTypes = null;

	public final static String LRE_CODE = "LRE";
	public final static String CERTIFIED_LETTER_CODE = "CERTIFIED_LETTER";
	
	public static void populateList(SessionClient session) throws JSONException, IOException, RemoteException {
		JSONObject deliveryAndLetterTypes = session.createApplicationClient().getDeliveryAndLetterTypes();
		populateList(deliveryAndLetterTypes.getJSONArray("letterDeliveryTypes"));
	}
	
	public static void populateList(JSONArray list) throws JSONException {
		letterDeliveryTypes = new ArrayList<LetterDeliveryType>();
		for (int i = 0; i < list.length(); i++) {
			JSONObject item = list.getJSONObject(i);
			LetterDeliveryType type = new LetterDeliveryType();
			type.id = item.getString("_id");
			type.code = item.getString("code");
			type.name = item.getString("name");
			letterDeliveryTypes.add(type);
		}
	}
	
	public static LetterDeliveryType getLetterDeliveryTypeById(String id) {
		for (LetterDeliveryType type : letterDeliveryTypes) {
			if (type.id.equals(id)) {
				return type;
			}
		}
		return null;
	}
	
	public static LetterDeliveryType getLetterDeliveryTypeByCode(String code) {
		for (LetterDeliveryType type : letterDeliveryTypes) {
			if (type.code.equals(code)) {
				return type;
			}
		}
		return null;
	}
	
	public String getId() {
		return id;
	}
	
	public String getCode() {
		return code;
	}
	
	public String getName() {
		return name;
	}
}

File: API/lb/api/beans/LetterDeliveryType.php
<?php


class LetterDeliveryType
{
	protected $id;
	protected $code;
	protected $name;

	protected static $letterDeliveryTypes = array();

	public static $LRE_CODE = "LRE";
	public static $CERTIFIED_LETTER_CODE = "CERTIFIED_LETTER";
	
	public static function populateList($applicationClient) {
		$deliveryAndLetterTypes = $applicationClient->getDeliveryAndLetterTypes();
		LetterDeliveryType::populateListByDeliveryTypeList($deliveryAndLetterTypes["letterDeliveryTypes"]);
	}


	
	public static function populateListByDeliveryTypeList($list) {
		LetterDeliveryType::$letterDeliveryTypes = array();
		
		for ($i = 0; $i < count($list); $i++) {
			$item = $list[$i];
			$type = new LetterDeliveryType();
			$type->id = $item["_id"];
			$type->code = $item["code"];
			$type->name = $item["name"];
			array_push(LetterDeliveryType::$letterDeliveryTypes, $type);
		}
		
	}
	
	public static function getLetterDeliveryTypeById($id) {
		for ($i = 0; $i < count(LetterDeliveryType::$letterDeliveryTypes); $i++) {
			$type = LetterDeliveryType::$letterDeliveryTypess[$i];
			if ($type->_id == $id) {
				return $type;
			}
		}
	}
	
	public static function getLetterDeliveryTypeByCode($code) {
		for ($i = 0; $i < count(LetterDeliveryType::$letterDeliveryTypes); $i++) {
			$type = LetterDeliveryType::$letterDeliveryTypes[$i];
			if ($type->code == $code) {
				return $type;
			}
		}
	}
	
	public function getId() {
		return $this->id;
	}
	
	public function getCode() {
		return $this->code;
	}
	
	public function getName() {
		return $this->name;
	}
}

?>